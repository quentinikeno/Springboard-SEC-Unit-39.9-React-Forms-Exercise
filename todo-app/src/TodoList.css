.TodoList {
	text-align: center;
}

.TodoList input {
	margin: 1rem;
	padding: 0.5rem;
}

.TodoList button {
	padding: 0.5rem;
	border: none;
	border-radius: 0.25rem;
	background-color: #405cf5;
	color: white;
	cursor: pointer;
}
