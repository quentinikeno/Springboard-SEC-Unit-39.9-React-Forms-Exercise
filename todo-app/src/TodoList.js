import { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
	const [todos, SetTodos] = useState([]);
	const addTodo = (newTodo) => {
		SetTodos((todos) => [...todos, newTodo]);
	};
	const editTodo = (id, updatedTask) => {
		SetTodos((todos) =>
			todos.map((todo) =>
				todo.id === id ? { ...todo, task: updatedTask } : todo
			)
		);
	};
	const setIsCompletedTodo = (id, isCompleted) => {
		SetTodos((todos) =>
			todos.map((todo) =>
				todo.id === id ? { ...todo, isCompleted } : todo
			)
		);
	};
	const removeTodo = (id) => {
		SetTodos((todos) => todos.filter((todo) => todo.id !== id));
	};

	const todoComponents = todos.map((todo) => (
		<Todo
			key={todo.id}
			id={todo.id}
			task={todo.task}
			isCompleted={todo.isCompleted}
			removeTodo={removeTodo}
			editTodo={editTodo}
			setIsCompletedTodo={setIsCompletedTodo}
		/>
	));
	return (
		<div className="TodoList">
			<h1>Todo List</h1>
			<NewTodoForm addTodo={addTodo} />
			<div className="TodoList-todos-div">{todoComponents}</div>
		</div>
	);
};

export default TodoList;
