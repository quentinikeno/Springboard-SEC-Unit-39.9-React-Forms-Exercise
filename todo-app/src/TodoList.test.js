import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

const addTodo = (todoList, task = "Feed the cat.") => {
	const taskInput = todoList.getByLabelText("Task");
	fireEvent.change(taskInput, { target: { value: task } });
	const submitButton = todoList.getByText("Add New Todo!");
	fireEvent.click(submitButton);
};

it("renders without crashing", () => {
	render(<TodoList />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

it("should add a new todo", () => {
	const todoList = render(<TodoList />);

	// No delete button should be in the DOM
	expect(todoList.queryByText("X")).not.toBeInTheDocument();

	// Add a new todo
	addTodo(todoList);

	// Task todo with delete button should be in the DOM
	expect(todoList.queryByText("Feed the cat.")).toBeInTheDocument();
	expect(todoList.queryByText("X")).toBeInTheDocument();
});

it("should remove a todo", () => {
	const todoList = render(<TodoList />);

	// Add a new todo
	addTodo(todoList);

	// Delete the new todo
	const removeButton = todoList.getByText("X");
	fireEvent.click(removeButton);

	// No delete button should be in the DOM
	expect(todoList.queryByText("X")).not.toBeInTheDocument();
});
