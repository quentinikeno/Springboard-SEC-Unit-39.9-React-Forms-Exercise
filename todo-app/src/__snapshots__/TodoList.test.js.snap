// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1>
      Todo List
    </h1>
    <form
      class="NewTodoForm"
    >
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        placeholder="Todo"
        type="text"
        value=""
      />
      <button>
        Add New Todo!
      </button>
    </form>
    <div
      class="TodoList-todos-div"
    />
  </div>
</DocumentFragment>
`;
